apps/api/src/app/workflows/usecases/create-workflow-override/create-workflow-override.usecase.ts

```diff
@@ -0,0 +1,85 @@
+import { BadRequestException, Injectable, NotFoundException } from '@nestjs/common';
+
+import {
+  NotificationTemplateEntity,
+  NotificationTemplateRepository,
+  TenantEntity,
+  TenantRepository,
+  WorkflowOverrideRepository,
+} from '@novu/dal';
+
+import { CreateWorkflowOverrideCommand } from './create-workflow-override.command';
+import { CreateWorkflowOverrideResponseDto } from '../../dto/create-workflow-override-response.dto';
+
+@Injectable()
+export class CreateWorkflowOverride {
+  constructor(
+    private tenantRepository: TenantRepository,
+    private notificationTemplateRepository: NotificationTemplateRepository,
+    private workflowOverrideRepository: WorkflowOverrideRepository
+  ) {}
+
+  async execute(command: CreateWorkflowOverrideCommand): Promise<CreateWorkflowOverrideResponseDto> {
+    const { tenant, workflow } = await this.extractEntities(command);
+
+    return await this.workflowOverrideRepository.create({
+      _environmentId: command.environmentId,
+      _tenantId: tenant._id,
+      _workflowId: workflow._id,
+      active: command.active,
+      preferenceSettings: command.preferenceSettings,
+    });
+  }
+
+  private async extractEntities(
+    command: CreateWorkflowOverrideCommand
+  ): Promise<{ tenant: TenantEntity; workflow: NotificationTemplateEntity }> {
+    const tenant = await this.tenantRepository.findOne({
+      _environmentId: command.environmentId,
+      identifier: command.tenantIdentifier,
+    });
+
+    if (!tenant) {
+      throw new NotFoundException(`Tenant with identifier ${command.tenantIdentifier} is not found`);
+    }
+
+    let workflow: NotificationTemplateEntity | null = null;
+
+    if (!command.triggerIdentifier && !command._workflowId) {
+      throw new BadRequestException(`Either triggerIdentifier or _workflowId must be provided`);
+    }
```

Comment:
maybe let's put it to the top, there is no sense in fetching the tenant and then rejecting the request

Reply:
Yes, agree this one is independent and go up :)